src/helpers/insights.ts

```diff
@@ -0,0 +1,55 @@
+import { InsightsClientMethod, InsightsClientPayload } from '../types';
+
+export function readDataAttributes(
+  domElement: HTMLElement
+): {
+  method: InsightsClientMethod;
+  payload: InsightsClientPayload;
+} {
+  const method: string | null = domElement.getAttribute('data-insights-method');
+  const serializedPayload: string | null = domElement.getAttribute(
+    'data-insights-payload'
+  );
+  if (typeof serializedPayload !== 'string') {
+    throw new Error(`The analytics helper expects payload to be a JSON string`);
+  }
+  try {
+    const payload = JSON.parse(serializedPayload);
+    return {
+      method: <InsightsClientMethod>method,
+      payload: <InsightsClientPayload>payload,
+    };
+  } catch (e) {
+    throw new Error('Unable to parse `data-insights-payload`');
+  }
+}
+
+export function hasDataAttributes(domElement: HTMLElement): boolean {
+  return domElement.hasAttribute('data-insights-method');
+}
+
+export function writeDataAttributes({
+  method,
+  payload,
+}: {
+  method: InsightsClientMethod;
+  payload: InsightsClientPayload;
+}) {
+  let serializedPayload;
```

Comment:
Since the error I want to catch needs to be only JSON serialisation related, I think it's best to keep only that instruction in the try block.
I'll move `if (typeof payload !== 'object')` to the first line of the function.